src/GitHub.App/Controllers/UIController.cs

```diff
@@ -253,13 +258,21 @@ public void Start([AllowNull] IConnection connection)
                     .IsLoggedIn(hosts)
                     .Do(loggedin =>
                     {
-                        if (!loggedin && currentFlow != UIControllerFlow.Authentication)
+                        if (currentFlow != UIControllerFlow.Authentication)
                         {
-                            connectionAdded = (s, e) => {
-                                if (e.Action == NotifyCollectionChangedAction.Add)
-                                    uiProvider.AddService(typeof(IConnection), e.NewItems[0]);
-                            };
-                            connectionManager.Connections.CollectionChanged += connectionAdded;
+                            if (loggedin) // register the first available connection so the viewmodel can use it
+                                uiProvider.AddService(connectionManager.Connections.First(c => hosts.LookupHost(c.HostAddress).IsLoggedIn));
```

Comment:
The `IsLoggedIn(hosts)` call that gives us the `loggedin` variable goes through all the `connectionManager.Connections` and attempts to log them in. If it returns true, it's because there's at least one connection on the list and that connection is logged in (it uses the same exact check I'm using here), so we can safely grab it.

I can do some bigger changes and instead of returning bool for the `Do()` block, I return the connection itself or null...